src/salaryAndBenefits/SalaryAndBenefits.tsx

```diff
@@ -1,16 +1,72 @@
+"use client"
+
 import styles from "./salaryAndBenefits.module.css";
 import Text from "src/components/text/Text";
 import {SalaryAndBenefits as SalaryAndBenefitsPayload} from "studio/lib/payloads/salaryAndBenefits";
 import Benefit from './components/benefit/Benefit';
+import { useState } from "react";
+import InputField from "src/components/forms/inputField/InputField";
+import { RadioButtonGroup } from "src/components/forms/radioButtonGroup/RadioButtonGroup";
+import { calculatePension, calculateSalary } from "src/salaryAndBenefits/utils/calculateSalary";
+
+interface IOption {
+  id: string;
+  label: string;
+  disabled?: boolean;
+  currentChecked: boolean;
+}
+
+const options: IOption[] = [
+  { id: "bachelor", label: "Bachelor", currentChecked: false, disabled: false },
+  { id: "master", label: "Master", currentChecked: true, disabled: false },
+];
+
 
 interface SalaryAndBenefitsProps {
   salaryAndBenefits: SalaryAndBenefitsPayload
 }
 
 const SalaryAndBenefits = ({salaryAndBenefits}: SalaryAndBenefitsProps) => {
+  const thisYear = new Date().getFullYear();
+  const [examinationYear, setExaminationYear] = useState(thisYear);
+  const [selectedOption, setSelectedOption] = useState("master");
+  const [salary, setSalary] = useState(
+    calculateSalary(thisYear, examinationYear, selectedOption)
+  );
+
+  const handleChange = (name: string, value: string) => {
+    if (name === "examinationYear") {
+      setExaminationYear(parseInt(value));
+      setSalary(calculateSalary(thisYear, parseInt(value), selectedOption));
```

Comment:
Could consider a `useMemo` for salary (or even just `const salary = calculateSalary(...)` instead of `useState` and `setSalary`

Reply:
I agreethat usememo is a better solution! Think we need this as degree and examinationyear could change